pkg/util/kubernetes/autoscalers/datadogexternal.go

```diff
@@ -141,6 +172,34 @@ func (p *Processor) queryDatadogExternal(metricNames []string) (map[string]Point
 	return processedMetrics, nil
 }
 
+func (p *Processor) updateRateLimiting() error {
```

Comment:
Just small nit: I think the current method name can be confusing. 
it only update "RateLimit" metrics, and not any logic parameter to call less the API in case of exhausted quota.

```suggestion
func (p *Processor) updateRateLimitingMetrics() error {
```